_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -1993,8 +2000,16 @@ void ImplementationAvc::OnEncodingQueried(DdiTaskIter task)
 
     if (m_useMbControlSurfs && task->m_isMBControl)
         ReleaseResource(m_mbControl, task->m_midMBControl);
-
-
+#if defined(MXF_ENABLE_MCTF_IN_AVC)
+    if (IsMctfSupported(m_video) && task->m_midMCTF)
+    {
+        ReleaseResource(m_mctf, task->m_midMCTF);
+        task->m_midMCTF = MID_INVALID
+            MfxHwH264Encode::Zero(task->m_handleMCTF);
```

Comment:
I don't understand what's written in these 2 lines. From my perspective this should not even be buildable.

Reply:
I need to release the MCTF output frame resource so I can reuse it, The check if MCTF is supported can go, because if MCTF is not enabled m_midMCTF will be zero always. After the resource is unlocked I then set the task pointer to MID_INVALID and zero the MCTF output handle in the task.

Reply:
Sorry, I realized that your comment was about the missing semi colon. I just fixed that.